leader-election-impl/src/test/java/com/palantir/leader/proxy/AwaitingLeadershipProxyTest.java

```diff
@@ -264,6 +267,53 @@ public void shouldClearDelegateUponLosingLeadership() throws Exception {
         verify(mock).close();
     }
 
+    @Test
+    public void whenMaybeNotCurrentLeaderAndLeadershipNotLostShouldNotClearDelegate() throws Exception {
+        MyCloseable mockA = mock(MyCloseable.class);
+        LeadershipCoordinator leadershipCoordinator = LeadershipCoordinator.create(leaderElectionService);
+        MyCloseable proxyA =
+                AwaitingLeadershipProxy.newProxyInstance(MyCloseable.class, () -> mockA, leadershipCoordinator);
+
+        // Wait to gain leadership
+        Uninterruptibles.sleepUninterruptibly(Duration.ofMillis(100L));
+
+        doThrow(new MaybeNotCurrentLeaderException(TEST_MESSAGE)).when(mockA).val();
+
+        assertThatThrownBy(proxyA::val)
+                .isInstanceOf(MaybeNotCurrentLeaderException.class)
+                .hasMessage(TEST_MESSAGE);
+        verify(mockA, times(0)).close();
+    }
```

Comment:
One more thing to test: do we want to check that the reference in proxyA still points to mockA? To do that, we can either call proxyA.val() again (catching the exception), or set mockA to do nothing, and then verify mockA had val called twice. (Or maybe some other way)

Reply:
The problem is, we will always return `mockA`, even if we create a new instance, as the supplier just returns the mock :/

Reply:
How about I just mock the supplier, and validate that it's only called once?

Reply:
(That's what I went with :D )

Reply:
Yeah that's good!